### ISSUE TITLE
Cannot tab back into an open floating OverflowMenu

### ISSUE THREAD

**Issue by nicthemighty** [carbon-design-system/carbon-components-react#2363]
## Environment

> Operating system

Mac osX

> Browser

Chrome v74.0

> Automated testing tool and ruleset

> Assistive technology used to verify

Keyboard navigation

## Detailed description

> What version of the Carbon Design System are you using?

10.1.0

> What did you expect to happen?

Could navigate via keyboard back into an open floating OverflowMenu

> What happened instead?

Tab focus never returns to the menu items

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Tab to menu trigger
2. Press space to open
3. Focus goes to primaryFocus item
4. Tab out of menu
5. Cannot tab back in

> Please create a reduced test case in CodeSandbox
> https://codesandbox.io/s/github/IBM/carbon-components-react/tree/master/examples/codesandbox

https://codesandbox.io/s/kon9z6o0rv

**Comment by asudoh** [comment]
Hi 👋 thank you for reporting - Here's the way you won't be lost with navigation. First, you can add `data-floating-menu-container` attribute to the containing `<div>`. Then you can add an focusable element next so such `<div>`. Your `App` component will be like:

```javascript
<>
  <div data-floating-menu-container>
    <OverflowMenu floatingMenu>
      <OverflowMenuItem primaryFocus>text1</OverflowMenuItem>
      <OverflowMenuItem>text2</OverflowMenuItem>
    </OverflowMenu>
  </div>
  <input type="text" class="bx--visually-hidden" />
</>
```

Hope this helps!

**Comment by nicthemighty** [comment]
@asudoh - Hi there, I have updated the code sandbox supplied and the issue is still persisting.

Could you confirm whether the data-xyz attributes can be used with the React components?

**Comment by asudoh** [comment]
Seems that your CodeBox hasn't been updated, the may explain why you are still seeing the problem. React JSX accepts `data-*` and `aria-*` with dash case. Once you replace the content with my markup I see the tab is not lost even if I tab away from the menu item.

**Comment by nicthemighty** [comment]
@asudoh - the code box appears to be updated when I or a colleague opens the link. I can see in the Chrome debugging tool that the data-overflow-menu-container attribute is in the HTML but it is not working for me.

Is there a minimum version of carbon-components or carbon-components-react I need?

**Comment by asudoh** [comment]
`data-overflow-menu-container` has been supported for long time. Let me clarify what I mentioned earlier a bit - When I put my markup into your CodeSandbox, the overflow menu started working in following way; Clicking on the overflow meun's trigger button puts the focus on the first item. Hitting tab then puts the focus back to the trigger button.

**Comment by nicthemighty** [comment]
@asudoh - Ah, thank you - it seems the additional empty `<>` wrapping the `data-overflow-menu-container`  `<div>` was the bit it needed to make it work. I'm assuming that adding `data-overflow-menu-container` to the root dom node was the problem?

Out of interest - I'm not seeing this in the react storybook. Is there a plan to update applicable storybook items to reference the need for the `data-*` attributes.

**Comment by asudoh** [comment]
@nicthemighty Additional `<input>` is not necessary for most applications, given it's very rare that the overflow menu is the last element in the document. However, you still have a good point and will consider better documentation.
